### ISSUE TITLE
Focus Indicator not returning to previous status from modal

### ISSUE THREAD

**Issue by israelcefrin** [pkp/pkp-lib#11258]
## Issue Description
When an editor opens a modal clicking on the `View`button from actions column, it opens the modal with the submission details. When they close the modal, the focus goes to the first interactive element of the page.

## How to reproduce / Current behavior
1) Log in to an OJS journal
2) Navigate using the TAB key to reach a `View` button
3) Activate the button using `ENTER` key
4) Close the modal pressing `ESC` key
5) Press `TAB` and see that the focus indicator is back to the journal name at the top

## What application are you using?
 OJS  version 3.5.rc2

## Stack used 
- Device: laptop
- OS: Macos Sequoia
- Browser: Edge
- Version: 135.0.3179.54 (Official build) (arm64)

## Testing method
- Method: Manual, Keyboard

## Action
When closing the modal, the focus should return to the button that triggered the modal.

## Additional information

**Original state of focus (on the View button)**
![Image](https://github.com/user-attachments/assets/48691b95-6b4a-4f71-ad2f-67440f11e201)

**Modal**

![Image](https://github.com/user-attachments/assets/19c5cfb9-21c8-4994-8584-d2990951ccb5)

**New focus position after closing the modal**
![Image](https://github.com/user-attachments/assets/d0fd8dd1-e202-43a1-b4a4-2ec72e2efb55)

**Comment by jardakotesovec** [comment]
Thanks @israelcefrin . 

For reference upstream issue: https://github.com/tailwindlabs/headlessui/issues/3686

**Comment by jardakotesovec** [comment]
Since the upstream is unresponsive, and currently with bunch of bugs. I tried to use reka-ui components for modals instead. Hopefully its going to be more perspective.

As result of that effort following should be improved:
- focus after closing modal should be moved the the element that opened the mdoal
- after opening modal - it either puts focus on the first item after the close button, or on the close button if there is no such item available. Note that for example workflow modal is fetching data before it shows the controls. Therefore in this case for example it puts focus on the close button first. Which might not be bad thing - it makes it easy to immediately close it while data are being fetched for example
- when multiple modals are opened, there was upstream bug that by clicking outside of the modal it was possible to close only top-most modal. Thats resolved with moving to new components
- for workflow modal - title is now set before data are fetched - so the screen reader has title to read. Most modal has title set when being opened. But this particular modal was relying on fetched data to show the title which was causing issue for screen reader


@Tribunal33 From testing point of view - things should look and work as before. Animations/styling is applied bit differently, so thats good to verify thats the same. Also some modal specific things like the issue we had with z-index and the quicksubmit plugin are good to check. Both side modal and the dialog are revamped



@israelcefrin will share VM details with you, so you can do some accessibility testing about these changes

@blesildaramirez Can you do the code review please?


This should also resolve this reported bug - https://github.com/pkp/pkp-lib/issues/11259

**Comment by Tribunal33** [comment]
The fix looks good.  I maintain the focus of the previous "View" submission path when I press `esc`. 

I did notice that Z-index issues might have cropped up for Quick Submit and Users & Roles. 
 
![Image](https://github.com/user-attachments/assets/48473e93-1423-4ccf-9a64-da87cdd20df7)

Repro steps > Users & Roles > Invite to a role > Search User > Cancel Invite
![Image](https://github.com/user-attachments/assets/c7ec785f-223d-4d09-bd9c-b5d802f3261a)

Oh one more location added for when Uploading a new plugin

![Image](https://github.com/user-attachments/assets/d71b363c-034c-4d03-813e-a2d6e0e42ea0)

**Comment by israelcefrin** [comment]
Hello @jardakotesovec ,

I've tested the interaction using keyboard only on : Edge, Firefox, Chrome, and Safari. The focus is returning to the element that opened the modal(s) successfully. 

During my tests, only Safari (MacOS) presented an issue. I couldn't open another modal because the focus wasn't moving to the links or buttons of the current modal. 
Using keyboard only
1. Open a modal from a  "view"  button
2. Move the focus to " activity log'" button and activate it pressing `ENTER` 
3. On the next modal press `TAB` key and notice that it doesn't move to the links

https://github.com/user-attachments/assets/92cab519-7512-4e09-bd6d-7e641eb638dc

**Comment by jardakotesovec** [comment]
@israelcefrin Thanks for the testing. I am on mac (macOS 15.5, safari 18.5) and works as expected. I tried multiple times - both locally and on the jarda.testing and works as expected for me.

I wonder whether some extension or some other version might cause this difference?

**Comment by israelcefrin** [comment]
Hello @jardakotesovec 

I've run a test on anonymous Firefox window and all good.  Also, I've run a test using Windows + Chrome + NVDA and the navigation works correctly with the interactions of open/close modal and returning to the trigger "view" button.

**Comment by Tribunal33** [comment]
@jardakotesovec I didn't have an issue but i'm on chrome.  Those z-index locations I pointed have, also, been fixed.

**Comment by jardakotesovec** [comment]
@Tribunal33 I still have some merging to do to 3.5 and main. :-)

**Comment by jardakotesovec** [comment]
Merged to 3.5 and main

@Tribunal33 Feel free to close it, depending if you want to do some additional testing.

**Comment by Tribunal33** [comment]
This specific bug passes QA on main.  I can return back to the previous focused element when returning from the submission view blade back to the submission list. However, there is a slight problem:

As noted in #11351 there is a difficulty in returning to focus when there are more levels of blade windows.  When choosing the 3 dots menu and another blade pops up and that window is closed the focus goes back to Submission List but current window open is Submission View.

**Comment by Tribunal33** [comment]
This passes QA. I have tested the most recent fix and the issue mentioned above has been addressed.  I have gone 6 levels deep for those side panels and it is accurately re-focusing the `TAB` into the appropriate section.  It is no longer defaulting back to the first main dashboard window.  @jardakotesovec once this is merged to main you can close it.

**Comment by Tribunal33** [comment]
This is now working on main and 3.5.  Closing.
